compiler.js

```diff
@@ -125,6 +129,42 @@ class SlangCompiler
         return disAsmCode;
     }
 
+    findDefinedEntryPoints(shaderSource)
+    {
+        var result = [];
+        try {
+            var slangSession = this.globalSlangSession.createSession(
+                this.compileTargetMap.findCompileTarget("SPIRV"));
+            if(!slangSession) {
+                return [];
+            }
+
+            var module = slangSession.loadModuleFromSource(shaderSource, "user", "/user.slang");
+            if(!module) {
+                return [];
+            }
+
+            var count = module.getDefinedEntryPointCount();
+            for (var i = 0; i < count; i++)
+            {
+                var entryPoint = module.getDefinedEntryPoint(i);
+                result.push(entryPoint.getName());
+                entryPoint.delete();
+            }
+        } catch (e) {
+            return [];
```

Comment:
suggest to add some log here.

Reply:
what log do you want to print? It is fine for this reflection API to return empty when there is an error, and I actually don't know why we will ever reach here.

Reply:
To my experience, If there is some bugs in the wasm code, bugs in slang code, or misused the binding APIs, it could go here.